pyo3-macros-backend/src/introspection.rs

```diff
@@ -216,10 +247,21 @@ impl IntrospectionNode<'_> {
                 content.push_tokens(if let Some(ident) = ident {
                     quote! { #ident::_PYO3_INTROSPECTION_ID }
                 } else {
-                    Ident::new("_PYO3_INTROSPECTION_ID", Span::call_site()).into_token_stream()
+                    quote! { _PYO3_INTROSPECTION_ID }
                 });
                 content.push_str("\"");
             }
+            Self::InputType {
+                rust_type,
+                nullable,
+            } => {
+                content.push_str("\"");
+                content.push_tokens(quote! { <#rust_type as #pyo3_crate_path::impl_::extract_argument::PyFunctionArgument<false>>::INPUT_TYPE });
+                if nullable {
+                    content.push_str(" | None");
```

Comment:
This would only support 3.10 and up. We _probably_ need to use `typing.Union` for now?

Reply:
Thinking about this further, I wonder if there's a way we can dynamically choose either `Union` or `|` syntax depending on the project's selected Python version?

Reply:
The `|` syntax [is recommanded](https://typing.python.org/en/latest/guides/writing_stubs.html#unions) even when targetting versions before 3.10. [Typeshed](https://github.com/python/typeshed) is making an heavy use of them. So, I guess it's fine to always use it. WDYT?

Reply:
Fair enough, probably in `.pyi` files it's ok given they aren't actually loaded by the interpreter 👍